src/Neo.VM/ReferenceCounter.cs

```diff
@@ -22,86 +22,180 @@ namespace Neo.VM
+
+            // Increment the item's stack references by the specified count.
             item.StackReferences += count;
-            zero_referred.Remove(item);
+
+            // Remove the item from the _zeroReferred set since it now has references.
+            _zeroReferred.Remove(item);
         }
 
+        /// <summary>
+        /// Adds an item to the zero-referred list.
+        ///
+        /// This method is used when an item has no remaining references.
+        /// It adds the item to the zero-referred list to be checked for cleanup later.
+        ///
+        /// Use this method when you detect that an item has zero references and may need to be cleaned up.
+        /// </summary>
+        /// <param name="item">The item to add.</param>
         internal void AddZeroReferred(StackItem item)
         {
-            zero_referred.Add(item);
+            // Add the item to the _zeroReferred set.
+            _zeroReferred.Add(item);
+
+            // If the item doesn't need to be tracked, return early.
             if (!NeedTrack(item)) return;
-            cached_components?.AddLast(new HashSet<StackItem>(ReferenceEqualityComparer.Instance) { item });
-            tracked_items.Add(item);
+
+            // Add the item to the cached components and the set of tracked items.
+            _cachedComponents?.AddLast(new HashSet<StackItem>(ReferenceEqualityComparer.Instance) { item });
+            _trackedItems.Add(item);
         }
 
+        /// <summary>
+        /// Checks and processes items that have zero references.
+        ///
+        /// This method is used to check items in the zero-referred list and clean up those that are no longer needed.
+        /// It uses Tarjan's algorithm to find strongly connected components and remove those with no references.
+        ///
+        /// Use this method periodically to clean up items with zero references and free up memory.
+        /// </summary>
+        /// <returns>The current reference count.</returns>
         internal int CheckZeroReferred()
         {
-            if (zero_referred.Count > 0)
+            // If there are items with zero references, process them.
+            if (_zeroReferred.Count > 0)
             {
-                zero_referred.Clear();
-                if (cached_components is null)
+                // Clear the zero_referred set since we are going to process all of them.
+                _zeroReferred.Clear();
+
+                // If cached components are null, we need to recompute the strongly connected components (SCCs).
+                if (_cachedComponents is null)
                 {
-                    //Tarjan<StackItem> tarjan = new(tracked_items.Where(p => p.StackReferences == 0));
-                    Tarjan tarjan = new(tracked_items);
-                    cached_components = tarjan.Invoke();
+                    // Create a new Tarjan object and invoke it to find all SCCs in the tracked_items graph.
```

Comment:
Next PR could be comment Tarjan class

Reply:
You can do that, it's too complex to understand.... I just assume it works fine. LOL

Reply:
EHUiaheuahea, I will ask @igormcoelho

Reply:
He knows this Tarjan algorithm very well, he teaches that.

Reply:
Awesome!!!!! Ask him the meaning of every single letter in that algorithm and comment them.